src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -79,7 +79,8 @@ public IDiscoveryRequest CreateDiscoveryRequest(DiscoveryCriteria discoveryCrite
             this.UpdateTestAdapterPaths(discoveryCriteria.RunSettings);
 
             var testHostManager = this.testHostProviderManager.GetTestHostManagerByRunConfiguration(discoveryCriteria.RunSettings);
-            
+            testHostManager.Initialize(TestSessionMessageLogger.Instance, discoveryCriteria.RunSettings);
```

Comment:
Please add unittest and acceptance test for discovering multiple dlls [here](https://github.com/Microsoft/vstest/blob/master/test/Microsoft.TestPlatform.AcceptanceTests/DiscoveryTests.cs#L14).